tests/test_zarr_ipfs.py

```
@@ -137,3 +137,84 @@ def test_encryption(random_zarr_dataset: tuple[str, xr.Dataset]):
+        transformer_encode=encrypt,
+        transformer_decode=decrypt,
+    )
+    test_ds.to_zarr(store=hamt, mode="w")
+
+    hamt.make_read_only()
+    loaded_ds = xr.open_zarr(store=hamt)
+    xr.testing.assert_identical(loaded_ds, expected_ds)
+
+
+    # Test with wrong API Key
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)